timeless/reservations/views.py

```diff
@@ -116,37 +117,44 @@ def list():
     return render_template("restaurants/tables/list.html")
 
 
-@BP.route("/create", methods=("GET", "POST"))
-def create():
-    """ Create new reservation """
-    """
-    @todo #380:30min Continue. When we use the form.validate() function the
-     status (enum: timeless/restaurants/models.py) is displaying the
-     following error: 'status': ['Invalid Choice: could not coerce', 'Not a
-     valid choice']. Removes the commented if to see the error. Also refactor
-     the function using genericViews, as explained in puzzles #134 and #137.
-    """
-    form = ReservationForm(request.form)
-    try:
-        """if request.method == "POST" and form.validate():"""
-        if request.method == "POST":
-            reservation = Reservation(
-                start_time=request.form["start_time"],
-                end_time=request.form["end_time"],
-                num_of_persons=request.form["num_of_persons"],
-                comment=request.form["comment"],
-                status=request.form["status"]
-            )
-            DB.session.add(reservation)
-            DB.session.commit()
-            return redirect(url_for("reservations.list"))
-        else:
-            flash("Error: ", form.errors)
-    except Exception as error:
-        flash("Error: ", error)
-
-    return render_template("reservations/create_edit.html", error=form.errors,
-                            action="create", form=form)
+class CreateReservation(views.CrudAPIView):
+    """ Create a new reservation instance """
+
+    def post(self):
+        """ Create new reservation """
+        """
+        @todo #434:30min Continue the implementation of CreateReservation. 
+         Add authentication and refactor using CrudAPIView also edit and 
+         delete methods for Reservations. Tests for these are skipped.
+         Created reservation should be returned as a JSON object
+        """
+        form = ReservationForm(request.form)
```

Comment:
@emilianodellacasa I guess forms do not work with json data correctly, need to setup https://marshmallow-sqlalchemy.readthedocs.io/en/latest/ to nicely work with json. also you should have used `request.get_json()` instead of `request.form` :)